src/Tools/dotnet-monitor/CollectionRules/Actions/ICollectionRuleAction.cs

```diff
@@ -13,6 +13,14 @@ internal interface ICollectionRuleAction
         Task StartAsync(CancellationToken token);
 
         Task<CollectionRuleActionResult> WaitForCompletionAsync(CancellationToken token);
+
+        CollectionRuleActionExecutionType ExecutionType { get; }
+    }
+
+    public enum CollectionRuleActionExecutionType
```

Comment:
This concept should not be necessary. The way you wrote the `LoadProfilerAction.ExecuteCoreAsync` method causes the execution system to wait for the profiler to be applied before considering the action startup to have been completed. So the target application will not be resumed, nor will any subsequent action start, before setting profiler has occurred.